docs/tasks/debug-application-cluster/core-metrics-pipeline.md

```diff
@@ -0,0 +1,47 @@
+---
+approvers:
+- fgrzadkowski
+- piosz
+title: Core metrics pipeline
+---
+
+Resource usage metrics are important for:
+
+- Kubernetes components to operate (e.g. autoscalers to automatically scale user applications)
+- Kubernetes users to understand their applications behavior
+
+In Kubernetes they are available through Metrics API.
+
+## Metrics API
+
+Metrics API provides current resource usage metrics for pods and nodes. It doesn't offer historical metrics, so one needs to
+archive metrics themselves. In the future Historical API might be defined but it will not be implemented as a part of
+the core metrics pipeline.
+
+The API is discoverable through the same endpoint as the other Kubernetes APIs under `/apis/metrics.k8s.io/` path.
+It also offers the same security, scalability and reliability guarantees.
+
+The API is defined in [k8s.io/metrics](https://github.com/kubernetes/metrics/tree/master/pkg/apis/metrics) repository,
+where the detailed semantic is explained.
+
+### Supported endpoints
+
+The list of supported endpoints in the API:
+
+- `/nodes` - all node metrics; support for `labelSelector` parameter
+- `/nodes/{node}` - metrics for the specified node
+- `/namespaces/{namespace}/pods` - all pod metrics within the namespace; support for `labelSelector` parameter
+- `/namespaces/{namespace}/pods/{pod}` - metrics for the specified pods
+
+## Metrics Server
+
+[Metrics Server](https://github.com/kubernetes-incubator/metrics-server) is a cluster-wide aggregator of monitoring data.
+Starting from Kubernetes 1.8 it's work on all Kubernetes setups
+and runs as a pod in a user space (on one of the nodes), similar to how any Kubernetes application would run.
+
+It discovers all nodes in the cluster and queries usage information from Summary API exposed by [Kubelet](/docs/admin/kubelet/)s.
```

Comment:
`Metric server collects metrics from the Summary API, exposed by [Kubelet](/docs/admin/kubelet/) on each node.`

Reply:
What is summary API? The link provided (https://kubernetes.io/docs/admin/kubelet/) is not providing a definition or example of this API.

Reply:
@crassirostris done

@tengqm I'm not sure whether there is any documentation for Summary API.
cc @dashpole @yujuhong

Reply:
do you mean the `metrics` endpoint https://<NODE_IP>:10250/metrics ?
I'm 100% sure the term "Summary API" never showed up anywehre on http://kubernetes.io/docs/

Reply:
I mean `https://<NODE_IP>:10255/stats/summary`

Reply:
> I'm not sure whether there is any documentation for Summary API.

Not that I know of...
Would provide the actual API help? https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/apis/stats/v1alpha1/types.go